lib/controllers/project.js

```diff
@@ -739,21 +729,63 @@ Project.prototype.packageProject = function(dir, includeModules, cb) {
   */
 //-----------------------------------------------------------------------------
 Project.prototype.detectProjectType = function (path, callback) {
+  var projectController = this;
+
   async.waterfall([
     function (next) {
-      Project.prototype.isMeteor(path, function (err, result) {
+      projectController.isMeteor(path, function (err, result) {
         if (err) return callback(err);
-        if (result) return callback(null, Project.types.METEOR);
-        else next();
+        if (result) return next('METEOR', true);
+        next();
       });
     }
-  ], function () {
-    callback(null, Project.types.NODEJS); // Default project type to Node.js
+  ], function (err) {
+    var result;
+
+    if (typeof err  !== 'object' && Project.types[err] !== 'undefined') {
+      result = err;
+      err = null;
+    }
+
+    callback(err, Project.types[result] || Project.types.NODEJS); // Default type to Node.js
   });
 };
 
 //-----------------------------------------------------------------------------
 /**
+  * Identifies if the source code version is supported by build/app servers.
+  * @param {String} srcVersion The version of the deploying application.
+  * @param {String} runtime The runtime of project being deployed.
+  * @param {Boolean} Returns boolean
+  */
+//-----------------------------------------------------------------------------
+Project.prototype.isSupported = function (srcVersion, runtime) {
```

Comment:
Can you add some tests for this? I think I see logic that won't work, would like to confirm:

```
isSupported('2.0.0', 'Meteor')
// return majorPass && minorPass && patchPass;
return true && false && true;
```